hudi-client/src/test/java/org/apache/hudi/metadata/TestHoodieMetadata.java

```diff
@@ -457,14 +479,14 @@ public void testArchivingAndCompaction() throws Exception {
     init(HoodieTableType.COPY_ON_WRITE);
 
     final int maxDeltaCommitsBeforeCompaction = 6;
+    // Test autoClean and asyncClean based on this flag which is randomly chosen.
+    boolean asyncClean = new Random().nextBoolean();
```

Comment:
this kind of randomization makes test non-deterministic. Please make this parameterized .